matlab/FeatherType.h

```diff
@@ -0,0 +1,119 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#ifndef MATLAB_ARROW_IPC_FEATHER_FEATHER_TYPE_H
+#define MATLAB_ARROW_IPC_FEATHER_FEATHER_TYPE_H
+
+#include <arrow/type.h>
+
+#include <matrix.h>
+
+namespace matlab {
+namespace arrow {
+namespace ipc {
+namespace feather {
+
+/// \brief A type traits class mapping Arrow types to MATLAB types.
+template <::arrow::Type::type ArrowTypeID>
+struct FeatherType;
```

Comment:
This could be named more intuitively. How about `MatlabTraits` or similar (we have `npy_traits` in the Python subpackage to get NumPy types from Arrow types).